### PR TITLE
serverless/client: Don't panic when tile is unreachable

### PR DESCRIPTION
**Pull request by Foxboron**
Happens if we specify a non-existing "to-size" in a consistency proof

Signed-off-by: Morten Linderud <morten@linderud.pw>

### CODE REVIEW

**Comment by AlCutter** [comment]
Thanks for the fix @Foxboron!

Incidentally, I'm curious how this happened - ideally the clients wanting the consistency proof should have signed checkpoints committing to both the smaller and larger sizes, so _in theory at least_ the nodes must have been written out. 

My guess would be something like the frontends serving the log state files are distributed and the state hadn't replicated to all frontends, so your client saw a newer/larger checkpoint from frontend A, but asked frontend B for tiles but B had not yet caught up?

**Comment by Foxboron** [comment]
A better check would be to just validate the input data for the known tree size instead of just claiming it's unreachable.

**Comment by AlCutter** [comment]
> A better check would be to just validate the input data for the known tree size instead of just claiming it's unreachable.

We could enforce that the `client` binary wants to verify the two `checkpoint` files before trying to build the proof between them, that would get rid of user error causing the issue but at a slight increase in complexity of having to keep those `checkpoint` artefacts around.  However the server-skew scenario can still easily happen so we'll always have the potential of a 404 looking for tile files...

File: serverless/client/client.go [hunk 1]
```diff
@@ -253,7 +253,11 @@ func (n *nodeCache) GetNode(ctx context.Context, id compact.NodeID) ([]byte, err
 		t = *tile
 		n.tiles[tKey] = *tile
 	}
-	node := t.Nodes[api.TileNodeKey(nodeLevel, nodeIndex)]
+	nodeKey := int(api.TileNodeKey(nodeLevel, nodeIndex))
+	if nodeKey > len(t.Nodes) {
+		return nil, fmt.Errorf("node %v (tile coords [%d,%d]/[%d,%d]) unreachable", id, tileLevel, tileIndex, nodeLevel, nodeIndex)
```
Comment (AlCutter) [inline serverless/client/client.go:258]: nit: `unreachable` -> `not found`?

**Review by AlCutter — APPROVED** [review]

**Comment by Foxboron** [comment]
Sorry for being slow, thanks for the fixup :)

**Pull request merged**